spec/flattenizerSpec.js

```diff
@@ -1,9 +1,60 @@
 'use strict';
-import {flatten, unflatten} from '../src/flattenizer';
+import { flatten, unflatten } from '../src/flattenizer';
 
 describe('Flattenator!', () => {
     describe('.flatten', () => {
 
+        it('will throw an error if a value passed in is not undef, but also not an object[string]', () => {
+            expect(() => {
+                flatten('boom')
+            }).toThrow(new TypeError('unflattened is not an object'));
+        });
+
+        it('will throw an error if a value passed in is not undef, but also not an object[number]', () => {
+            expect(() => {
+                flatten(5)
+            }).toThrow(new TypeError('unflattened is not an object'));
+        });
+
+        it('will throw an error if a value passed in is not undef, but also not an object[function]', () => {
+            expect(() => {
+                flatten(() => {return  'why would you do this...'})
```

Comment:
Actually, I may be wrong since this anonymous function is passed in as a function param. I did some brief investigation in dev console and got weird results:
```
let bar = (f) => { console.log(f) }
undefined
bar(() => 'hi')
() => 'hi'
undefined
```